x/emissions/keeper/keeper.go

```diff
@@ -826,6 +835,16 @@ func (k *Keeper) AppendInference(ctx context.Context, topicId TopicId, nonce typ
 		newInferences.Inferences = append(newInferences.Inferences[:lowScoreIndex], newInferences.Inferences[lowScoreIndex+1:]...)
 		newInferences.Inferences = append(newInferences.Inferences, inference)
 		return k.allInferences.Set(ctx, key, newInferences)
+	} else {
+		// Update EMA score using previous active set score at topic-defined quantile among the last active set of inferers
+		topic, err := k.GetTopic(ctx, topicId)
+		if err != nil {
+			return err
+		}
+		err = k.CalcAndSaveInfererScoreEmaWithLastSavedTopicQuantile(ctx, topic, block, inference.Inferer)
```

Comment:
if this is the N-th time the inferer is sending inference in same block (yes, paying for it), seems like this will this apply EMA over and over with the quantile on this same block. Seems like this should happen only once per block to avoid people paying for cleaning its own EMA.

Reply:
See my latest comment/convo with @relyt29